elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/Context.java

```diff
@@ -73,197 +58,165 @@
+            return UNRESOLVED;
+        }
+
+        @Override
+        public Object resolve(Object context) {
+            return null;
         }
 
-        if (argsIndex >= 0) {
-            Map<String, ? extends Object> pinnedArgs = getArgumentMapFromString(column.substring(argsIndex));
-            invokedColumnName = column.substring(0, argsIndex);
-            return invokedCtx.get(invokedColumnName, pinnedArgs);
+        @Override
+        public Set<Entry<String, Object>> propertySet(Object context) {
+            return null;
         }
-        return invokedCtx.get(invokedColumnName);
     }
 
     /**
-     * Resolves the handlebars with in expression.
-     * @param columnName Name of the column whose default arguments to be used for resolving this expression.
-     * @param expression Expression to resolve.
-     * @return fully resolved expression.
+     * This tells compiler how to write a {@link StringValue} object as String.
      */
-    public String resolveHandlebars(String columnName, String expression) {
-        ColumnProjection column = new ColumnProjection() {
-            @Override
-            public <T extends ColumnProjection> T withProjected(boolean projected) {
-                return null;
+    @AllArgsConstructor
+    @Getter
+    protected class StringValue {
+        private final String value;
+
+        @Override
+        public String toString() {
+            // Sql helper returns the value already enclosed inside '{{' and '}}'
+            if (value.startsWith(HANDLEBAR_PREFIX)) {
+                return value;
             }
-            @Override
-            public Pair<ColumnProjection, Set<ColumnProjection>> nest(Queryable source, MetaDataStore store,
-                            boolean joinInOuter) {
-                return null;
-            }
-            @Override
-            public ValueType getValueType() {
-                return null;
-            }
-            @Override
-            public String getName() {
-                return columnName;
-            }
-            @Override
-            public String getExpression() {
-                return expression;
-            }
-            @Override
-            public ColumnType getColumnType() {
-                return null;
-            }
-        };
-
-        return resolveHandlebars(column);
+            return HANDLEBAR_PREFIX + value + HANDLEBAR_SUFFIX;
+        }
     }
 
-    public static Map<String, Object> getDefaultArgumentsMap(Set<Argument> availableArgs) {
+    /**
+     * If get method returns a {@link StringValue} object. This class tell how to resolve a field
+     * against {@link StringValue} object.
+     * eg: To resolve {{join.col1}}, get method is called with key 'join' first,
+     *     This will return a {@link StringValue} object which wraps the value 'join'.
+     *     Now 'col1' is resolved against this {@link StringValue} object.
+     */
+    protected class StringValueResolver implements ValueResolver {
```

Comment:
Another idea that might simplify this.

What if {{join.col1}} returned a map that is populated:

join returns a Map with 'col1'
'col1' returns : {{join.col1}}

This should be simpler to understand than trying to coerce handlebars to do something here.